src/services.h

```diff
@@ -55,33 +58,51 @@
 namespace dd
 {
   /* service types as variant type. */
-  typedef mapbox::util::variant<MLService<CaffeLib,ImgCaffeInputFileConn,SupervisedOutput,CaffeModel>,
+  typedef mapbox::util::variant<
+#ifdef USE_CAFFE
+    MLService<CaffeLib,ImgCaffeInputFileConn,SupervisedOutput,CaffeModel>,
     MLService<CaffeLib,CSVCaffeInputFileConn,SupervisedOutput,CaffeModel>,
     MLService<CaffeLib,TxtCaffeInputFileConn,SupervisedOutput,CaffeModel>,
     MLService<CaffeLib,SVMCaffeInputFileConn,SupervisedOutput,CaffeModel>,
     MLService<CaffeLib,ImgCaffeInputFileConn,UnsupervisedOutput,CaffeModel>,
     MLService<CaffeLib,CSVCaffeInputFileConn,UnsupervisedOutput,CaffeModel>,
     MLService<CaffeLib,TxtCaffeInputFileConn,UnsupervisedOutput,CaffeModel>,
+    MLService<CaffeLib,SVMCaffeInputFileConn,UnsupervisedOutput,CaffeModel>
+#endif
 #ifdef USE_CAFFE2
+    #ifdef USE_CAFFE
```

Comment:
```
#ifndef PREV_SERVICE
    #define PREV_SERVICE
#else
    ,
#endif
```
With this kind of directive, the MLServices can be position-independent in the code